corehq/apps/es/aggregations.py

```diff
@@ -0,0 +1,251 @@
+"""
+Aggregate Queries
+---------------
+Aggregations are a replacement for Facets
+
+Here is an example used to calculate how many new pregnancy cases each user has
+opened in a certain date range.
+
+.. code-block:: python
+
+    res = (CaseES()
+           .domain(self.domain)
+           .case_type('pregnancy')
+           .date_range('opened_on', gte=startdate, lte=enddate))
+           .aggregation(TermsAggregation('by_user', 'opened_by')
+           .size(0)
+
+    buckets = res.aggregations.by_user.buckets
+    buckets.user1.doc_count
+
+There's a bit of magic happening here - you can access the raw json data from
+this aggregation via ``res.aggregation('by_user')`` if you'd prefer to skip it.
+
+The ``res`` object has a ``aggregations`` property, which returns a namedtuple
+pointing to the wrapped aggregation results.  The name provided at instantiation is
+used here (``by_user`` in this example).
+
+The wrapped ``aggregation_result`` object has a ``result`` property containing the
+aggregation data, as well as utilties for parsing that data into something more
+useful. For example, the ``TermsAggregation`` result also has a ``counts_by_bucket``
+method that returns a ``{bucket: count}`` dictionary, which is normally what you
+want.
+
+As of this writing, there's not much else developed, but it's pretty easy to
+add support for other facet types and more results processing
+"""
```

Comment:
Great job @snopoke!  I appreciate that you made this feel a lot like the rest of ESQuery.